### PR TITLE
Lint: Attempt to lint everything

### PR DESCRIPTION
**Pull request by TheCharlatan**

### CODE REVIEW

File: src/common/plugin/makeCurrencyTools.ts [hunk 1]
```diff
@@ -83,18 +83,19 @@ export function makeCurrencyTools(
...
       // Parse the pathname and add it to the result object
-      if (pathname) {
+      if (pathname !== '') {
         if (currencyInfo.currencyType === EngineCurrencyType.UTXO) {
           const parsedPath = utxoUtils.parsePathname({
             pathname: uriObj.pathname,
             coin: currencyInfo.network,
             network: NetworkEnum.Mainnet
           })
+          // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
```
Comment (thehobbit85) [inline src/common/plugin/makeCurrencyTools.ts:98]: test for 'parsedPath == null'

File: src/common/utxobased/db/Models/ProcessorTransaction.ts [hunk 2]
```diff
@@ -41,6 +41,7 @@ export const toEdgeTransaction = async (
       async (i: string) =>
         args.walletTools.scriptPubkeyToAddress({
           scriptPubkey: args.tx.outputs[parseInt(i)].scriptPubkey,
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```
Comment (thehobbit85) [inline src/common/utxobased/db/Models/ProcessorTransaction.ts:44]: Try and change this function to not be a 'one liner' and then you might be able to fix the linting problem.

File: src/common/utxobased/db/makeProcessor.ts [hunk 3]
```diff
@@ -167,7 +167,8 @@ async function createOrOpen<T extends BaseType>(
     }
   } catch (err) {
     console.log('already exists', config)
-    if (!err.message.includes('already exists')) {
+    // eslint-disable-next-line no-extra-boolean-cast
+    if (!Boolean(err.message.includes('already exists'))) {
```
Comment (thehobbit85) [inline src/common/utxobased/db/makeProcessor.ts:171]: No need to cast to Boolean but you might need to actually check that 'message' is in 'err'

File: src/common/utxobased/db/makeProcessor.ts [hunk 4]
```diff
@@ -247,6 +248,7 @@ export async function makeProcessor(
 
     // If there is path info on the address to save but not stored in
     // the previously existing data, save to the by path database.
+    // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
```
Comment (thehobbit85) [inline src/common/utxobased/db/makeProcessor.ts:251]: check 'addressData?.path != null'

File: src/common/utxobased/db/makeProcessor.ts [hunk 5]
```diff
@@ -391,6 +395,7 @@ export async function makeProcessor(
     const [address] = await baselets.addressByScriptPubkey.query('', [
       scriptPubkey
     ])
+    // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
```
Comment (thehobbit85) [inline src/common/utxobased/db/makeProcessor.ts:398]: != null

File: src/common/utxobased/db/makeProcessor.ts [hunk 6]
```diff
@@ -412,7 +417,8 @@ export async function makeProcessor(
       address.lastTouched = data.lastTouched
     }
 
-    if (data.used && !address.used) {
+    // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
```
Comment (thehobbit85) [inline src/common/utxobased/db/makeProcessor.ts:420]: compare to 'null'

File: src/common/utxobased/engine/makeSpendHelper.ts [hunk 7]
```diff
@@ -15,6 +15,7 @@ export function calculateFeeRate(
     spendTargets
   } = spendInfo
 
+  // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
```
Comment (thehobbit85) [inline src/common/utxobased/engine/makeSpendHelper.ts:18]: compare to 'null'

File: src/common/utxobased/engine/makeUtxoEngine.ts [hunk 8]
```diff
@@ -248,13 +246,15 @@ export async function makeUtxoEngine(
       const addressData = await processor.fetchAddressByScriptPubkey(
         scriptPubkey
       )
+      // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
```
Comment (thehobbit85) [inline src/common/utxobased/engine/makeUtxoEngine.ts:249]: use 'ternary' instead of the '!!'. 
I actually like using '!!' but we want to make typescript happy :)

File: src/common/utxobased/engine/makeUtxoEngineState.ts [hunk 9]
```diff
@@ -419,6 +418,7 @@ const internalGetFreshAddress = async (
   let scriptPubkey = await processor.fetchScriptPubkeyByPath(path)
   scriptPubkey =
     scriptPubkey ?? (await walletTools.getScriptPubkey(path)).scriptPubkey
+  // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
```
Comment (thehobbit85) [inline src/common/utxobased/engine/makeUtxoEngineState.ts:421]: compare to 'null'

File: src/common/utxobased/engine/makeUtxoEngineState.ts [hunk 10]
```diff
@@ -428,58 +428,23 @@ const internalGetFreshAddress = async (
   })
 }
 
-interface ProcessFormatAddressesArgs extends FormatArgs {}
-
-const processFormatAddresses = async (args: ProcessFormatAddressesArgs) => {
```
Comment (thehobbit85) [inline src/common/utxobased/engine/makeUtxoEngineState.ts:433]: Is that not used?

File: src/common/utxobased/engine/makeUtxoEngineState.ts [hunk 11]
```diff
@@ -518,6 +483,7 @@ const processAddressTransactions = async (
 
   // If address is used and previously not marked as used, mark as used.
   const used = txs > 0 || unconfirmedTxs > 0
+  // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
```
Comment (thehobbit85) [inline src/common/utxobased/engine/makeUtxoEngineState.ts:486]: compare to 'null'

File: src/common/utxobased/engine/makeUtxoEngineState.ts [hunk 12]
```diff
@@ -526,6 +492,7 @@ const processAddressTransactions = async (
 
   for (const rawTx of transactions) {
     const tx = processRawTx({ ...args, tx: rawTx })
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
```
Comment (thehobbit85) [inline src/common/utxobased/engine/makeUtxoEngineState.ts:495]: add a 'catch'

File: src/common/utxobased/engine/makeUtxoWalletTools.ts [hunk 13]
```diff
@@ -69,8 +69,9 @@ export function makeUtxoWalletTools(
   // Convert xprivs to xpubs
   const xpubKeys = Object.assign({}, xprivKeys)
   for (const key in xpubKeys) {
-    const format = <CurrencyFormat>key
+    const format = key as CurrencyFormat
     xpubKeys[format] = xprivToXPub({
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```
Comment (thehobbit85) [inline src/common/utxobased/engine/makeUtxoWalletTools.ts:74]: use '??' or 'ternary' to fix the lint issue

File: src/common/utxobased/engine/makeUtxoWalletTools.ts [hunk 14]
```diff
@@ -81,6 +82,7 @@ export function makeUtxoWalletTools(
   const fns: UTXOPluginWalletTools = {
     getPubkey(args: AddressPath): string {
       return xpubToPubkey({
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```
Comment (thehobbit85) [inline src/common/utxobased/engine/makeUtxoWalletTools.ts:85]: use '??' or 'ternary' to fix the lint issue

File: src/common/utxobased/engine/makeUtxoWalletTools.ts [hunk 15]
```diff
@@ -128,6 +130,7 @@ export function makeUtxoWalletTools(
 
     getPrivateKey(args: AddressPath) {
       return xprivToPrivateKey({
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```
Comment (thehobbit85) [inline src/common/utxobased/engine/makeUtxoWalletTools.ts:133]: use '??' or 'ternary' to fix the lint issue

File: src/common/utxobased/engine/paymentRequest.ts [hunk 16]
```diff
@@ -55,6 +55,7 @@ export async function getPaymentDetails(
 
   const edgePaymentProtocolInfo: EdgePaymentProtocolInfo = {
     nativeAmount: `${nativeAmount}`,
+    // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
```
Comment (thehobbit85) [inline src/common/utxobased/engine/paymentRequest.ts:58]: Try adding 'JSON.stringify' around each param to see if it fix the lint error

File: src/common/utxobased/engine/utils.ts [hunk 17]
```diff
@@ -215,6 +215,7 @@ export const deriveXpub = (args: {
 }): string =>
   xprivToXPub({
     ...args,
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```
Comment (thehobbit85) [inline src/common/utxobased/engine/utils.ts:218]: Change this to be a bit 'safer' and actually check the arguments

File: src/common/utxobased/keymanager/base.ts [hunk 18]
```diff
@@ -25,6 +25,7 @@ export function base58Base(
     const newChecksum = checksumFn(payload)
 
     if (
+      // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
```
Comment (thehobbit85) [inline src/common/utxobased/keymanager/base.ts:28]: Compare to '1'

File: src/common/utxobased/keymanager/bitcoincashUtils/base32.ts [hunk 19]
```diff
@@ -20,6 +20,7 @@ interface CHARSET_INVERSE_INDEX {
   [value: string]: number
 }
 
+// eslint-disable-next-line @typescript-eslint/no-redeclare
```
Comment (thehobbit85) [inline src/common/utxobased/keymanager/bitcoincashUtils/base32.ts:23]: change the name of the type 'CHARSET_INVERSE_INDEX'

File: src/common/utxobased/keymanager/bitcoincashUtils/bn.ts [hunk 20]
```diff
@@ -22,13 +22,15 @@ BN.Minus1 = new BN(-1)
 
 BN.fromNumber = function (n: number) {
   if (typeof n !== 'number') {
+    // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
```
Comment (thehobbit85) [inline src/common/utxobased/keymanager/bitcoincashUtils/bn.ts:25]: Try adding 'JSON.stringify' around each param to see if it fix the lint error

File: src/common/utxobased/keymanager/bitcoincashUtils/bn.ts [hunk 21]
```diff
@@ -22,13 +22,15 @@ BN.Minus1 = new BN(-1)
...
     throw new Error(`InvalidArgument: ${n} not a number`)
   }
   return new BN(n)
 }
 
 BN.fromString = function (str: string, base: number) {
   if (typeof str !== 'string') {
+    // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
```
Comment (thehobbit85) [inline src/common/utxobased/keymanager/bitcoincashUtils/bn.ts:33]: Try adding 'JSON.stringify' around each param to see if it fix the lint error

File: src/common/utxobased/keymanager/bitcoincashUtils/cashAddress.ts [hunk 22]
```diff
@@ -71,6 +71,7 @@ const convertBits = (
   for (let i = 0; i < data.length; i++) {
     const value = data[i]
     if (value < 0 || value >> from !== 0) {
+      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
```
Comment (thehobbit85) [inline src/common/utxobased/keymanager/bitcoincashUtils/cashAddress.ts:74]: Try adding 'JSON.stringify' around each param to see if it fix the lint error

File: src/common/utxobased/network/BlockBook.ts [hunk 23]
```diff
@@ -221,6 +207,7 @@ export function makeBlockBook(config: BlockBookConfig): BlockBook {
             if (response.data?.subscribed === true) {
               return
             }
+            // eslint-disable-next-line @typescript-eslint/no-floating-promises
```
Comment (thehobbit85) [inline src/common/utxobased/network/BlockBook.ts:210]: Add 'catch' to promise

File: src/common/utxobased/network/Socket.ts [hunk 24]
```diff
@@ -74,6 +74,7 @@ function setupBrowser(
   uri: string,
   callbacks?: InnerSocketCallbacks
 ): InnerSocket {
+  // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
```
Comment (thehobbit85) [inline src/common/utxobased/network/Socket.ts:77]: Compare with 'null'

File: src/common/utxobased/network/Socket.ts [hunk 25]
```diff
@@ -105,6 +106,7 @@ function setupBrowser(
 
     disconnect() {
       if (
+        // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
```
Comment (thehobbit85) [inline src/common/utxobased/network/Socket.ts:109]: Compare with 'null'

File: src/common/utxobased/network/Socket.ts [hunk 26]
```diff
@@ -128,6 +130,7 @@ function setupWS(uri: string, callbacks?: InnerSocketCallbacks): InnerSocket {
   })
 
   ws.on('message', (data: WS.Data) => {
+    // eslint-disable-next-line @typescript-eslint/no-base-to-string
```
Comment (thehobbit85) [inline src/common/utxobased/network/Socket.ts:133]: Try casting into String

File: src/common/utxobased/network/Socket.ts [hunk 27]
```diff
@@ -151,6 +154,7 @@ function setupWS(uri: string, callbacks?: InnerSocketCallbacks): InnerSocket {
     },
 
     disconnect(): void {
+      // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
```
Comment (thehobbit85) [inline src/common/utxobased/network/Socket.ts:157]: Compare with 'null'

File: src/types.ts [hunk 28]
```diff
@@ -2,6 +2,7 @@ import { EdgeCorePlugins } from 'edge-core-js/lib/types/types'
 
 declare global {
   interface Window {
+    // eslint-disable-next-line @typescript-eslint/method-signature-style
```
Comment (thehobbit85) [inline src/types.ts:5]: change to:   addEdgeCorePlugins?: (plugins: EdgeCorePlugins) => void

File: test/common/utxobased/db/Processor.spec.ts [hunk 29]
```diff
@@ -26,19 +18,6 @@ describe('Processor', function () {
   const emitter = new EventEmitter() as any
   let processor: Processor
 
-  const format: CurrencyFormat = 'bip44'
```
Comment (thehobbit85) [inline test/common/utxobased/db/Processor.spec.ts:29]: Why is this removed?

**Review by thehobbit85 — CHANGES_REQUESTED** [review]

File: src/common/utxobased/db/makeProcessor.ts (hunk 4 above)
Comment (TheCharlatan) [reply src/common/utxobased/db/makeProcessor.ts:251]: This is weird either way, if I read this right, we only reach this far if addressData is indeed null.

Comment (TheCharlatan) [reply src/common/utxobased/db/makeProcessor.ts:251]: This is weird either way, if I am reading it right, addressData is anyway null.

File: src/common/utxobased/engine/makeUtxoEngineState.ts (hunk 10 above)
Comment (TheCharlatan) [reply src/common/utxobased/engine/makeUtxoEngineState.ts:433]: Apparently no longer, removed here: https://github.com/EdgeApp/edge-currency-plugins/commit/0794a357bc0459cb878d53aeabcb801643442689#diff-01dd589479440f34b8666c05923b1c9a88a8446d3a6bdeb058e7795d49eb5fd4L100

File: test/common/utxobased/db/Processor.spec.ts (hunk 29 above)
Comment (TheCharlatan) [reply test/common/utxobased/db/Processor.spec.ts:29]: It's not used.
